src/merkle/merkle-objects.ts

```diff
@@ -202,10 +202,13 @@ export class IpfsMerge implements MergeFunction<CIDHolder, TreeMetadata> {
 
   async merge(
     left: Node<CIDHolder>,
-    right: Node<CIDHolder>,
+    right: Node<CIDHolder> | null,
     metadata: TreeMetadata | null
   ): Promise<Node<CIDHolder>> {
-    const merged = [left.data.cid, right.data.cid]
+    const merged = [left.data.cid]
```

Comment:
```suggestion
    const merged = [left?.data?.cid, right?.data?.cid]
```

Reply:
I think this simplifies things and lets you remove the extra `merged.push` line

Reply:
nice

Reply:
oh wait nevermind, this will add result in `undefined` not `null`